cdap-tms/src/main/java/co/cask/cdap/messaging/store/TopicMetadataCodec.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.messaging.store;
+
+import co.cask.cdap.internal.guava.reflect.TypeParameter;
+import co.cask.cdap.internal.guava.reflect.TypeToken;
+import co.cask.cdap.messaging.TopicMetadata;
+import co.cask.cdap.proto.id.TopicId;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Codec class to serialize/deserialize {@link TopicMetadata}.
+ */
+public class TopicMetadataCodec implements JsonSerializer<TopicMetadata>, JsonDeserializer<TopicMetadata> {
```

Comment:
`TopicMetadata` only has simple fields. We can just mark fields that we don't want to serialize as `transient` (e.g. `validated`).